iotedgedev/utility.py

```diff
@@ -188,3 +189,22 @@ def nested_set(self, dic, keys, value):
             current = current.get(key)
 
         current[keys[-1]] = value
+
+    @staticmethod
+    @suppress_all_exceptions()
+    def hash_connection_str_hostname(connection_str_hostname):
+        """Hash connection string hostname to count distint IoT Hub number"""
+
+        # get hostname suffix (e.g., azure-devices.net) to distinguish national clouds
+        if "." in connection_str_hostname:
+            hostname_suffix = connection_str_hostname[connection_str_hostname.index(".")+1:]
+        else:
+            hostname_suffix = ""
+
+        return (Utility.get_sha256_hash(connection_str_hostname), hostname_suffix)
```

Comment:
Same with another PR, we expect to record empty string when host name is empty.

Reply:
Updated. Please help check again.